interfaces/builtin/basedeclaration.go

```diff
@@ -283,7 +283,6 @@ slots:
         - app
     deny-connection: true
   modem-manager:
-    deny-connection: true
```

Comment:
I realize you did this to work around https://bugs.launchpad.net/bugs/1640874, but this must be removed since the base declaration was working as intended (ie, it is meant to require a snap declaration for connections to be allowed to modem-manager).